fundingmanager_test.go

```diff
@@ -520,24 +520,63 @@ func openChannel(t *testing.T, alice, bob *testNode, localFundingAmt,
 	return fundingOutPoint
 }
 
-func assertMarkedOpen(t *testing.T, alice, bob *testNode,
-	fundingOutPoint *wire.OutPoint) {
-	state, _, err := alice.fundingMgr.getChannelOpeningState(fundingOutPoint)
-	if err != nil {
-		t.Fatalf("unable to get channel state: %v", err)
+func assertNumPendingChannels(t *testing.T, node *testNode, expectedNum int) {
+	var numPendingChans int
+	for i := 0; i < 10; i++ {
+		pendingChannels, err := node.fundingMgr.cfg.Wallet.Cfg.Database.FetchPendingChannels()
+		if err != nil {
+			t.Fatalf("unable to fetch pending channels: %v", err)
+		}
+
+		numPendingChans = len(pendingChannels)
+		if numPendingChans == expectedNum {
+			// Success, return.
+			return
+		}
+
+		// Sleep, and try again in a bit.
+		time.Sleep(500 * time.Millisecond)
```

Comment:
do we need to sleep on the last iteration? another alternative is to sleep at the top of the loop if only if i > 0